packages/block-editor/src/components/block-navigation/branch.js

```diff
@@ -1,59 +1,100 @@
+/**
+ * External dependencies
+ */
+import { map, compact } from 'lodash';
+
 /**
  * WordPress dependencies
  */
-import { Children, cloneElement, useContext } from '@wordpress/element';
-import { Fill, Slot } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import BlockNavigationListItem from './list-item';
-import { BlockNavigationContext } from './list';
-import { BlockListBlockContext } from '../block-list/block';
+import BlockNavigationBlock from './block';
+import BlockNavigationAppender from './appender';
 
-const BlockNavigationBranch = ( { children, ...props } ) => {
-	const { __experimentalWithBlockNavigationSlots } = useContext(
-		BlockNavigationContext
-	);
-	if ( ! __experimentalWithBlockNavigationSlots ) {
-		return (
-			<li>
-				<BlockNavigationListItem { ...props } />
-				{ children }
-			</li>
-		);
-	}
+export default function BlockNavigationBranch( props ) {
```

Comment:
@adamziel I ended up renaming `BlockNavigationBranch` to `BlockNavigationLeaf` and `BlockNavigationList` to `BlockNavigationBranch`, so this diff looks a bit confusing.

I think that's correct though, a branch should have multiple leaves, and the leaf should be the terminating points on a tree.